src/main/java/com/eternalcode/parcellockers/user/UserRepositoryJdbcImpl.java

```diff
@@ -0,0 +1,134 @@
+    }
+
+    @Override
+    public CompletableFuture<Optional<User>> findByName(String name) {
+        return CompletableFuture.supplyAsync(() -> {
+            try (ResultSet resultSet = this.jdbcConnectionProvider.executeQuery("SELECT * FROM `users` WHERE `name` = ? LIMIT 1".replace("?", name))) {
+                return this.extractUser(resultSet);
+            }
+            catch (SQLException exception) {
+                throw new RuntimeException(exception);
+            }
+        });
+    }
+
+    @Override
+    public CompletableFuture<Optional<User>> findByUuid(UUID uuid) {
+        return CompletableFuture.supplyAsync(() -> {
+            try (ResultSet resultSet = this.jdbcConnectionProvider.executeQuery("SELECT * FROM `users` WHERE `uuid` = ? LIMIT 1".replace("?", uuid.toString()))) {
+                return this.extractUser(resultSet);
+            }
+            catch (SQLException exception) {
+                throw new RuntimeException(exception);
+            }
+        });
+    }
+
+    @Override
+    public CompletableFuture<List<User>> findAll() {
+        return CompletableFuture.supplyAsync(() -> {
+            Set<Parcel> parcelSet = this.parcelRepository.findAll().join();
+            List<User> users = new ArrayList<>();
+
+            try (ResultSet resultSet = this.jdbcConnectionProvider.executeQuery("SELECT * FROM `users`")) {
+
+                while (resultSet.next()) {
+                    Set<UUID> userParcels = new HashSet<>();
+
+                    for (Parcel parcel : parcelSet) {
+                        UUID target = UUID.fromString(resultSet.getString("uuid"));
+
+                        if (parcel.getSender().equals(target)) {
+                            userParcels.add(parcel.getUuid());
+                        }
+                    }
+                    User user = new User(UUID.fromString(resultSet.getString("uuid")), resultSet.getString("name"), userParcels);
+
+                    users.add(user);
+                }
+            }
+            catch (SQLException exception) {
+                throw new RuntimeException(exception);
+            }
+
+            return users;
+        });
+    }
+
+    @Override
+    public CompletableFuture<Void> remove(User user) {
+        return CompletableFuture.runAsync(() ->
+                this.jdbcConnectionProvider.executeUpdate("DELETE FROM `users` WHERE `uuid` = ?".replace("?", user.getUuid().toString())))
+                .orTimeout(5, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public CompletableFuture<Void> remove(UUID uuid) {
+        return CompletableFuture.runAsync(() ->
+                this.jdbcConnectionProvider.executeUpdate("DELETE FROM `users` WHERE `uuid` = ?".replace("?", uuid.toString())))
+                .orTimeout(5, TimeUnit.SECONDS);
+    }
+
+    @NotNull
+    private Optional<User> extractUser(ResultSet resultSet) throws SQLException {
```

Comment:
czemu to się nazywa extractUser a nie po prostu get?

Reply:
Bo ta metoda powstała po to, aby oddzielić zduplikowany kod